bson/mgocompat/registry.go

```diff
@@ -86,9 +86,10 @@ func newMgoRegistryBuilder() *bsoncodec.RegistryBuilder {
 	return rb
 }
 
-// buildMgoRNVRegistry creates a new bsoncodec.RegistryBuilder configured to behave like mgo/bson with RespectNilValues set to true.
-func newMgoRespectNilValuesRegistryBuilder() *bsoncodec.RegistryBuilder {
-	rb := newMgoRegistryBuilder()
+// NewMgoRespectNilValuesRegistryBuilder creates a new bsoncodec.RegistryBuilder configured to behave like mgo/bson
+// with RespectNilValues set to true.
+func NewMgoRespectNilValuesRegistryBuilder() *bsoncodec.RegistryBuilder {
```

Comment:
What is the default behaviour of `mgo` with respect to this?

Reply:
mgo's default behavior is RespectNilValues=false